docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,872 @@
+  - [monitor.js](#monitorjs)
+    - [Environment Variables](#envvarsmonitor)
+    - [Metrics](#metrics)
+    - [readOracleAddresses()](#readOracleAddresses)
+    - [readMonitoringState()](#readMonitoringState)
+    - [getOraclesInvitations()](#getOraclesInvitations)
+    - [updateMetrics(oracleName, oracle, feed, value, id, actualPrice, lastRound)](#updateMetrics)
+    - [updateBalanceMetrics(oracleName, oracle, feed, value)](#updateBalanceMetrics)
+    - [queryPrice(feed)](#queryPriceMonitor)
+    - [getOffersAndBalances(follower, oracle)](#getOffersAndBalances)
+    - [getLatestPrices(oracle, oracleDetails, state)](#getLatestPrices)
+    - [startServer()](#startServer)
+    - [monitor()](#monitorfunc)
+- [Considerations](#considerations)
+  - [Why not use a Chainlink cron job?](#consid1)
+  - [Why does the middleware consist of an exposed API and two intervalled functions?](#consid2)
+  - [Why does the middleware only consider the last few offers when making checks?](#consid3)
+  - [Intervals](#consid4)
+## Oracle Network Structure
+
+<img src="images/struct.png"></img>
+<br>
+
+Each node operator has to run:
+- A Chainlink node hosted locally in Docker that queries Chainlink adapters
+- Chainlink adapters hosted beside the Chainlink node that act as middleware between the node as various 3rd party APIs
+- Middleware which initiates jobs on the Chainlink node through its API if a round hasn't been created in 1 minute on the on chain aggregator contract. Rounds are also created if the oracle operator's current price is deviated past X% from the latest on-chain median price. Oracle operators are not allowed to create 2 rounds subsequently to prevent spamming.
+- An Agoric node hosted on their own infrastructure to allow the middleware / oracle proxy to communicate and broadcast transactions reliably to the network.
+
+
+## Smart Contract Details
+
+Smart Contract - https://github.com/Agoric/agoric-sdk/blob/master/packages/inter-protocol/src/price/priceAggregatorChainlink.js
+
+- The smart contract resembles a native Chainlink Solidity Flux-Monitor smart contract on Ethereum.
+- For each feed, there will be a smart contract on chain and each node operator receives invitations to their smart wallet to be part of the oracle set. Then, oracles can use that invitation to push prices on chain.
+- There is a minimum number of submissions which has to be reached in order for the on-chain price to be updated
+- An oracle cannot initialise multiple consecutive rounds to avoid network congestion
+- The updated price is the median from the submitted prices for a particular round
+- Submissions for old rounds cannot be submitted and will be rejected
+
+## Aims/Objectives
+
+#### Middleware
+
+The middleware is needed to:
+1. Create CL jobs. Chainlink Jobs are created every minute. A cron job is not used in order to be able to pass in particular inputs to the job run so that the progress of the request can be stored and monitored.
```

Comment:
Consistency, CL jobs or Chainlink Jobs pick one

Reply:
What does this mean? "A cron job is not used in order to be able to pass in particular inputs to the job run so that the progress of the request can be stored and monitored." so the program is running all the time?